growthbook/growthbook
# Add version string comparison support to the conditions UI

### Features and Changes

Adds some UI to support doing version string comparison targeting conditions.

- Closes https://linear.app/growthbook/issue/GB-262/[fe]-version-string-comparison-ui

### Dependencies



### Testing

- Add a new string attribute `version`
- Create a new feature flag of any type. In the screenshots below I created `greeting` of type string
- Create a new forced rule and choose one of the "version is X" options (see screenshots below)
- This should populate your features JSON properly after publishing

You can see an example of this implemented [here](https://github.com/growthbook/examples/pull/37).


### Screenshots

<img width="1126" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/aed62fb6-9386-4893-9f9d-d1e428beb4da">

UI for "greater than" and other comparison looks the same but under the hood uses the version string comparison operators.

<img width="857" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/d8dc7906-b0bf-42d0-9eb4-6e4b2730f2ef">

<img width="825" alt="image" src="https://github.com/growthbook/growthbook/assets/113377031/65ef238d-caaa-4746-84c9-13d5dec88be1">


<img width="1410" alt="sdk attribute format" src="https://github.com/growthbook/growthbook/assets/113377031/45f4dda5-a8f6-4abc-b2b2-e03e0a433a3f">


![CleanShot 2023-05-31 at 18 03 06@2x](https://github.com/growthbook/growthbook/assets/113377031/5368111b-eaa4-4eb9-8d20-e468045d80ad)
